operations/rack_resiliency/rack_resiliency.md

```diff
@@ -0,0 +1,489 @@
+
+A similar problem emerges for utility storage on management storage nodes. Even though these are not orchestrated by `Kubernetes', incorrect physical placement and/or a lack of correct software configuration, specifically around replication, may cause utility storage service disruptions due to rack-level failures.
+
+### Key features
+
+* Validates and reports node placement configuration in order for admin to reconfigure, if required, where critical management services can tolerate rack failures.
+* Provides tolerance for a single rack failure within the Management Plane Failure Domain (MPFD), ensuring that it does not lead to a system-wide outage.
+* Ensures load balancing and high availability of critical services essential for the seamless execution and completion of user jobs.
+* Monitor and report failures of management node(s) and rack(s) which are part of MPFD
+* Monitor and report critical services health and their balance across racks
+* Minimize operational complexity for admins based on above key features
+* Aligns with future plans for similar functionality in next generation systems management solutions
+* Ease of deploying and managing
+  
+## RR solution overview
+
+The architecture for providing a Rack Resiliency solution includes the following 4 stages, along with a CLI for managing RR.
+
+* Management plane placement discovery and placement validation
+* Spreading of NCNs across `Kubernetes` zones, and spreading storage nodes across `Ceph` zones 
+* Distribute critical services across zones
+* Monitor failures and ensure critical services are available across zones
+* RR CLI
+
+![Rack Resiliency solution overview](../../img/rack-resiliency-1.png)
+
+###  1 Discover management plane and validate for resiliency 
+
+#### 1.1 Identify racks and management/ managed nodes across racks
+
+The discovery of the physical racks and management nodes (Master, Worker and Storage) in Cray Ex system happens with the "Discovery" method, where during the node personalization stage of booting a management node, the discovery and validation operation is executed. Hardware State Manager (HSM) REST APIs are used to get component details for segregating management nodes across racks in the form of xnames. 
+
+SLS (System Layout Services) REST APIs are used to map these management nodes xnames to the corresponding `Kubernetes` and storage nodes names. This mapping of rack to `Kubernetes` and storage nodes is stored in the below format as a JSON file to be consumed by the `Kubernetes` and `Ceph` zoning modules later.
+
+**Example of node mapping JSON file**
+
+`rr_hw_discovery.json`
+
+```text
+{
+x3000:[ncn-m001,ncn-w001,ncn-w004,ncn-w007,ncn-s001],
+x3001:[ncn-m002,ncn-w002,ncn-w006,ncn-w005,ncn-w008],
+x3002:[ncn-m003,ncn-w003,ncn-w009,ncn-s005,ncn-s002]
+}
+```
+
+#### 1.2 Validate distribution of management/ managed nodes

```

Comment:
Does this actually validate the distribution of the managed nodes? Or does it just print informational messages about their presence (as the text below seems to indicate)? If it is the latter, the header should probably be revised. In that case, I'd just take the reference to managed nodes out of the header entirely, since that's clearly not the purpose of this step. It's worth mentioning below that it is mentioned informationally, but I think it's misleading to say their distribution is validate, if it's only mentioning which racks they are in.